### PR TITLE
Added Bicycle Power device profile, and some refactoring

### PR DESCRIPTION
**Pull request by David-Hari**
I added a class to process messages from the Bicycle Power device, initially only supporting 2 of the 5 available data pages.

I also refactored some of the `ant.plus` code so that device profiles can share common code in a super class called `DeviceProfile`. This includes things like pairing, searching and closing. I felt that the separate `_EventHandler` class was unnecessary, so I moved that code into `DeviceProfile`.

Another change I made was to replace the class-based callbacks with a dictionary of functions. I felt this was easier to use, and allows one to specify the same function as a callback for multiple devices (say, for handling time-outs).

### CODE REVIEW

File: src/ant/plus/heartrate.py [hunk 1]
```diff
@@ -29,61 +29,27 @@
...
-        """Called when heart rate data is received.
-
-        computed_heartrate - computed heart rate from monitor
-        event_time_ms - event time in ms
-        rr_interval_ms - time between successive R waves
+        :param node: The ANT node to use
+        :param network: The ANT network to connect on
+        :param callbacks: Dictionary of string-function pairs specifying the callbacks to
+                use for each event. In addition to the events supported by `DeviceProfile`,
+                `BicyclePower` also has the following:
```
Comment (mch) [inline src/ant/plus/heartrate.py]: Seems like BicyclePower should be HeartRate here.

File: src/ant/plus/stride.py [hunk 2]
```diff
@@ -29,48 +29,29 @@
...
-    """
-
-    def __init__(self, node, network, device_id=0, transmission_type=0, callback=None):
-        """TODO
-
+        :param node: The ANT node to use
+        :param network: The ANT network to connect on
+        :param callbacks: Dictionary of string-function pairs specifying the callbacks to
+                use for each event. In addition to the events supported by `DeviceProfile`,
+                `BicyclePower` also has the following:
```
Comment (mch) [inline src/ant/plus/stride.py]: Seems like BicyclePower should be Stride here.

**Review by mch — APPROVED** [review]
I had been converting names toward PEP 8, so `wraparound_difference` instead of `wraparoundDifference`, and this goes in the other direction. The original code base I forked was trending towards camelCase as well. If camelCase is common in Python code bases I'm cool with it, but otherwise I'd move toward PEP 8. That can be discussed more and addressed later though. 

Thanks for taking care of a huge number of TODO's! 

I had been trying to increase covered of stuff in core to support refactoring, but I don't think any of your changes are risky so no increase in coverage there is fine. 

I haven't had a chance to run the tests yet. I will do that sometime this weekend and then merge this.

File: src/ant/plus/heartrate.py (hunk 1 above)
Comment (David-Hari) [reply src/ant/plus/heartrate.py]: Yes, your right. Damn copy paste!

**Comment by David-Hari** [comment]
> I had been converting names toward PEP 8, so wraparound_difference instead of wraparoundDifference, and this goes in the other direction. The original code base I forked was trending towards camelCase as well. If camelCase is common in Python code bases I'm cool with it, but otherwise I'd move toward PEP 8. That can be discussed more and addressed later though.

Yeah, I was keeping the naming consistent with the rest of the code-base.

**Pull request merged**